src/editor/mod.rs

```diff
@@ -69,6 +71,7 @@ impl Editor {
             RedrawEvent::ModeInfoSet { cursor_modes } => self.cursor.mode_list = cursor_modes,
             RedrawEvent::OptionSet { gui_option } => self.set_option(gui_option),
             RedrawEvent::ModeChange { mode_index } => {
+                self.mode = mode_index;
```

Comment:
Maybe a more principled change, although somewhat larger, would be to store the mode info list in the editor, store the current mode also in the editor and set the mode object itself in the cursor object.

This way we have access to the current mode name, and it pushes the editor state up to the editor instead of storing that in the cursor like its done today.

Reply:
will have to fiddle around more to get this done, it'll also benefit us in the long run if we ever come to doing more fancy stuff like status line theming etc